spec/models/spree/promotion_handler/cart_spec.rb

```diff
@@ -0,0 +1,30 @@
+require 'spec_helper'
+
+module Spree
+  module PromotionHandler
+    describe Cart, type: :model do
+      let(:line_item) { create(:line_item) }
+      let(:order) { line_item.order }
+
+      let(:promotion) { Promotion.create(name: "At line items") }
+      let(:multi_promotion) { create :promotion, multi_coupon: true }
+      let(:calculator) { Calculator::FlatPercentItemTotal.new(preferred_flat_percent: 10) }
+
+      subject { Cart.new(order, line_item) }
+
+      context '#promotions' do
```

Comment:
@AnuragBhardwaj1 Let's write the specs for the public methods where `promotions` is being used.